UI/NpcList/NpcList.gd

```diff
@@ -6,21 +6,95 @@ onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
 onready var childrenButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/ChildrenAmount
+onready var popupWindow = $CenterContainer/Notification
+onready var popupWindowLabel = $CenterContainer/Notification/NotificationLabel
+onready var popupForgetBtn = $CenterContainer/Notification/HBoxC/Forget
+onready var popupOkBtn = $CenterContainer/Notification/HBoxC/Ok
+onready var popupCancelBtn = $CenterContainer/Notification/HBoxC/Cancel
 var _nameBtnState: bool = true
 var _genderBtnState: bool = true
 var _personalityBtnState: bool = true
 var _childrenBtnState: bool = true
+var _IDtoForget
+var nodeToFree
 
 
-func addRow(name: String, gender: String, subbyStat: float, ID: String, children: int = 0):
+func addRow(name: String, gender: String, subbyStat: float, ID: String, occupation: String, children: int = 0):
 	var newRow = npcRow.instance()
 	container.add_child(newRow)
-	newRow.initData(name, gender, subbyStat, ID, children)
+	newRow.initData(name, gender, subbyStat, ID, occupation, children)
 	newRow.connect("onForgetButtonPressed", self, "forgetNPC")
+	newRow.connect("onMeetButtonPressed", self, "meetNPC")
+	
+
+func forgetNPC(ID, name, node):
+	_IDtoForget = ID
+	nodeToFree = node
+	sendPopupMessage("Are you sure you want to forget "+name+"? This will permanently remove "+name+" from the game. Keep in mind that if this character is pregnant, their pregnancy will be forgotten too. But any kids you had together will stay.", true)
+
+
+func _on_Forget_pressed():
+	nodeToFree.queue_free()
+	nodeToFree = null
+	GM.main.removeDynamicCharacter(_IDtoForget)
+	hideNotificationButnsAndWindow()
+
+
+func _on_Cancel_pressed():
+	hideNotificationButnsAndWindow()
+
+
+func meetNPC(ID, occupation):
+	var room = GM.world.getRoomByID(GM.pc.getLocation())
+	match occupation:
+		"Inmates": 
+			if(WorldPopulation.Inmates in GM.pc.getLocationPopulation()):
+				GM.main.runScene("InmateExposureForcedSexScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no inmates in this location.\nTry looking elswhere")
+		"Guards":
+			if(WorldPopulation.Guards in GM.pc.getLocationPopulation()):
+			#if(room.loctag_GuardsEncounter || room.loctag_Greenhouses):
+				GM.main.runScene("GuardCaughtOfflimitsScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no guards in this location.\nTry looking at the checkpoint or near greenhouses")
+		"Engineers":
+			if(room.loctag_EngineersEncounter):
+				GM.main.runScene("EngineerCaughtOfflimitsScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no engineers in this location.\nTry looking at the engineering bay")
+		"Nurses":
+			if(room.loctag_MentalWard):
+				GM.main.runScene("NurseCaughtOfflimitsScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no nurses in this location.\nTry looking at the restricted area of the mediacl ward")
```

Comment:
typo `mediacl`
`There are no nurses in this location.\nTry searching in the restricted area of the medical ward` would sound better